source/libNsdl/src/sn_grs.c

```diff
@@ -887,7 +875,14 @@ sn_nsdl_dynamic_resource_parameters_s *sn_grs_search_resource(struct grs_s *hand
         /* Scan all nodes on list */
         ns_list_foreach(sn_nsdl_dynamic_resource_parameters_s, resource_search_temp, &handle->resource_root_list) {
             /* If length equals.. */
-            if (resource_search_temp->static_resource_parameters->pathlen == pathlen) {
+            size_t len = 0;
+            if(resource_search_temp &&
+                    resource_search_temp->static_resource_parameters &&
+                    resource_search_temp->static_resource_parameters->path) {
```

Comment:
Are those "resource_search_temp && resource_search_temp->static_resource_parameters" checks really needed? The resource_search_temp should never be null here if I remember ns_list_foreach correctly, and the static_resource_parameters should be validated when a new item is added to list, not afterwards.